### PR TITLE
Improvements for Reactive REST client

### PR DESCRIPTION
**Pull request by fedinskiy**

### CODE REVIEW

File: QUARKUS-1415.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+Test plan covers RESTEasy Reactive functionality provided by `quarkus-resteasy-reactive` and `quarkus-rest-client-reactive` extensions,
+with emphasis on multipart form data support.
+
+https://quarkus.io/version/main/guides/resteasy-reactive
+
+https://quarkus.io/version/main/guides/rest-client-reactive
+
+## Related test plans
+- QUARKUS-1075
```
Comment (rsvoboda) [inline QUARKUS-1415.md:13]: Make it clickable / link

File: QUARKUS-1415.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
+# QUARKUS-1415 Reactive REST Client improvements
+
+JIRA link: https://issues.redhat.com/browse/QUARKUS-1415
+
+Test plan covers RESTEasy Reactive functionality provided by `quarkus-resteasy-reactive` and `quarkus-rest-client-reactive` extensions,
+with emphasis on multipart form data support.
+
+https://quarkus.io/version/main/guides/resteasy-reactive
+
+https://quarkus.io/version/main/guides/rest-client-reactive
```
Comment (rsvoboda) [inline QUARKUS-1415.md:10]:
```suggestion
Documentation:
 - https://quarkus.io/version/main/guides/resteasy-reactive
 - https://quarkus.io/version/main/guides/rest-client-reactive
```

File: QUARKUS-1415.md [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+## Related test plans
+- QUARKUS-1075
+
+## Scope of the testing
+- Upload and download of large files
+- Support for proxy authentication
+- New configuration schema(https://github.com/quarkusio/quarkus/pull/17220)
```
Comment (rsvoboda) [inline QUARKUS-1415.md:18]:
```suggestion
- New configuration schema introduced in https://github.com/quarkusio/quarkus/pull/17220
```

File: QUARKUS-1415.md [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+https://quarkus.io/version/main/guides/resteasy-reactive
+
+https://quarkus.io/version/main/guides/rest-client-reactive
+
+## Related test plans
+- QUARKUS-1075
+
+## Scope of the testing
+- Upload and download of large files
```
Comment (rsvoboda) [inline QUARKUS-1415.md:16]: `large file` is what exactly, please define concrete sizes?

File: QUARKUS-1415.md [hunk 5]
```diff
@@ -0,0 +1,28 @@
+# QUARKUS-1415 Reactive REST Client improvements
+
+JIRA link: https://issues.redhat.com/browse/QUARKUS-1415
+
+Test plan covers RESTEasy Reactive functionality provided by `quarkus-resteasy-reactive` and `quarkus-rest-client-reactive` extensions,
```
Comment (rsvoboda) [inline QUARKUS-1415.md:5]: [QUARKUS-1415](https://issues.redhat.com/browse/QUARKUS-1415) is about Reactive REST Client improvements

File: QUARKUS-1415.md [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+https://quarkus.io/version/main/guides/resteasy-reactive
+
+https://quarkus.io/version/main/guides/rest-client-reactive
+
+## Related test plans
+- QUARKUS-1075
+
+## Scope of the testing
+- Upload and download of large files
+- Support for proxy authentication
```
Comment (rsvoboda) [inline QUARKUS-1415.md:17]: How do you plan to simulate proxy?

File: QUARKUS-1415.md [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+with emphasis on multipart form data support.
+
+https://quarkus.io/version/main/guides/resteasy-reactive
+
+https://quarkus.io/version/main/guides/rest-client-reactive
+
+## Related test plans
+- QUARKUS-1075
+
+## Scope of the testing
```
Comment (rsvoboda) [inline QUARKUS-1415.md:15]: These topics are not covered:
 - support for ParamConverter
 - recursive sub-resources support, that is sub-sub-resources and so on work too,
 - UI / dev experience - List of clients in Dev UI and Links to client interfaces (if in user's app)

https://resteasy.dev/2021/04/05/blog-ParamConverter-with-Quarkus/ can be used for ParamConverter coverage

sub-resources can be a bit tricky topic, we should make sure one can't end up with infinite loop of sub-resources

UI/dev experience should be checked as part of the initial experiments, not necessary to de TD for it, but should be checked at least manually though

File: QUARKUS-1415.md [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+- Upload and download of large files
+- Support for proxy authentication
+- New configuration schema(https://github.com/quarkusio/quarkus/pull/17220)
+
+## Impact on test suites and testing automation:
+- Four new scenarios in QE TS, approximate time increase should be around several minutes, mostly due to uploading and downloading of big files
```
Comment (rsvoboda) [inline QUARKUS-1415.md:21]: In native the impact will be bigger, my guess is 4x ~4 minutes

**Review by rsvoboda — CHANGES_REQUESTED** [review]

File: QUARKUS-1415.md [hunk 9]
```diff
@@ -0,0 +1,30 @@
...
+
+Documentation:
+https://quarkus.io/version/main/guides/resteasy-reactive
+https://quarkus.io/version/main/guides/rest-client-reactive
+
+## Related test plans
+- [QUARKUS-1075](QUARKUS-1075.md)
+
+## Scope of the testing
+- Upload and download of large(more than 2Gb) files, see https://github.com/quarkusio/quarkus/pull/22229 for details
```
Comment (rsvoboda) [inline QUARKUS-1415.md:15]: Is it realistic to fetch 2 GB of data using REST Client?

Comment (rsvoboda) [inline QUARKUS-1415.md:15]:
```suggestion
- Upload and download of large (more than 250MB) files, see https://github.com/quarkusio/quarkus/pull/22229 for details
```

File: QUARKUS-1415.md (hunk 6 above)
Comment (rsvoboda) [reply QUARKUS-1415.md:16]: Any concrete proxy? If so please include its name / image tag here

**Review by rsvoboda — CHANGES_REQUESTED** [review]

File: QUARKUS-1415.md (hunk 6 above)
Comment (fedinskiy) [reply QUARKUS-1415.md:16]: Nothing in particular. Any recommendations?

File: QUARKUS-1415.md (hunk 9 above)
Comment (fedinskiy) [reply QUARKUS-1415.md:15]: The original issue[1] explicitly mentions "files larger, than 2G". File of size 250MB will not really validate it.

[1] https://github.com/quarkusio/quarkus/issues/22218

Comment (fedinskiy) [reply QUARKUS-1415.md:15]: It is, but it will take some time(between 10 and 30 minutes, I guess)

File: QUARKUS-1415.md (hunk 6 above)
Comment (rsvoboda) [reply QUARKUS-1415.md:16]: No, you are preparing the plan, so I was wondering if you have something.

File: QUARKUS-1415.md (hunk 9 above)
Comment (rsvoboda) [reply QUARKUS-1415.md:15]:
```suggestion
- Upload and download of large (more than 2GB) files, see https://github.com/quarkusio/quarkus/pull/22229 for details
```

let's see if native will be able to handle 2GB+ size :)

File: QUARKUS-1415.md [hunk 10]
```diff
@@ -0,0 +1,30 @@
...
+- Support for proxy authentication
+- New configuration schema introduced in https://github.com/quarkusio/quarkus/pull/17220
+- Recursive sub-resources (see https://github.com/quarkusio/quarkus/pull/22417)
+- Support for ParamConverter (related issue: https://github.com/quarkusio/quarkus/issues/23486)
+- Manual check of Dev UI 
+
+## Impact on test suites and testing automation:
+- Four new scenarios in QE TS, approximate time increase should be around several minutes, mostly due to uploading and downloading of big files. Additional impact for native is expected to be around 4 minutes
```
Comment (rsvoboda) [inline QUARKUS-1415.md:23]:
```suggestion
- Four new scenarios in QE TS
- Upload and download of large (more than 2GB) files will add several minutes to the execution time
- Additional impact for Native mode is expected to be around 4 minutes per new scenario
- Overall time execution increase is ~20 minutes
```

**Review by rsvoboda — CHANGES_REQUESTED** [review]

**Pull request merged**